modules/apps/journal/journal-web/src/test/java/com/liferay/journal/web/internal/util/DataDefinitionUtilTest.java

```diff
@@ -0,0 +1,180 @@
+		_fieldReference = RandomTestUtil.randomString();
+	}
+
+	@Test
+	public void testValidateDefinitionFields() {
+		_testValidateDefinitionFieldsWithExistingDataDefinition();
+		_testValidateDefinitionFieldsWithExistingInvalidName();
+		_testValidateDefinitionFieldsWithNewDataDefinition();
+	}
+
+	private DataDefinition _getDataDefinition() {
+		DataDefinition dataDefinition = new DataDefinition();
+
+		dataDefinition.setDefaultDataLayout(_getDataLayout(_fieldName));
+
+		DataDefinitionField dataDefinitionField = new DataDefinitionField();
+
+		dataDefinitionField.setName(_fieldName);
+
+		dataDefinitionField.setCustomProperties(
+			HashMapBuilder.<String, Object>put(
+				"fieldReference", _fieldReference
+			).build());
+
+		dataDefinition.setDataDefinitionFields(
+			new DataDefinitionField[] {dataDefinitionField});
+
+		return dataDefinition;
+	}
+
+	private String _getDataDefinitionFieldName(DataDefinition dataDefinition) {
+		DataDefinitionField dataDefinitionField =
+			dataDefinition.getDataDefinitionFields()[0];
+
+		return dataDefinitionField.getName();
+	}
+
+	private DataLayout _getDataLayout(String fieldName) {
+		DataLayout dataLayout = new DataLayout();
+
+		dataLayout.setDataLayoutPages(
+			new DataLayoutPage[] {
+				new DataLayoutPage() {
+					{
+						dataLayoutRows = new DataLayoutRow[] {
+							new DataLayoutRow() {
+								{
+									dataLayoutColumns = new DataLayoutColumn[] {
+										new DataLayoutColumn() {
+											{
+												fieldNames = new String[] {
+													fieldName
+												};
+											}
+										}
+									};
+								}
+							}
+						};
+						description = HashMapBuilder.<String, Object>put(
+							"en_US", "Description"
+						).build();
+						title = HashMapBuilder.<String, Object>put(
+							"en_US", "Title"
+						).build();
+					}
+				}
+			});
+
+		return dataLayout;
+	}
+
+	private DDMStructure _getDDMStructure(String existingFieldName) {
+		DDMStructure ddmStructure = Mockito.mock(DDMStructure.class);
+		DDMForm ddmForm = Mockito.mock(DDMForm.class);
+
+		Mockito.when(
+			ddmStructure.getDDMForm()
+		).thenReturn(
+			ddmForm
+		);
+
+		DDMFormField ddmFormField = new DDMFormField(existingFieldName, "text");
+
+		ddmFormField.setFieldReference(_fieldReference);
+
+		Mockito.when(
+			ddmForm.getDDMFormFieldsReferencesMap(true)
+		).thenReturn(
+			Map.of(_fieldReference, ddmFormField)
+		);
+
+		return ddmStructure;
+	}
+
+	private void _testValidateDefinitionFieldsWithExistingDataDefinition() {
+		DataDefinition dataDefinition = _getDataDefinition();
+
+		String existingFieldName = DDMFormFieldUtil.getDDMFormFieldName(
+			_fieldName);
+
+		DataDefinitionUtil.validateDefinitionFields(
+			dataDefinition, _getDDMStructure(existingFieldName));
+
+		String newFieldName = _getDataDefinitionFieldName(dataDefinition);
+
+		Assert.assertEquals(existingFieldName, newFieldName);
+
+		Assert.assertTrue(DataDefinitionUtil.isValidFieldName(newFieldName));
```

Comment:
@gergelyszalay, here it would be better to first assertTrue and then assertEquals